lib/parseMarkdown.js

```diff
@@ -12,33 +12,112 @@ const {
   whole,
   wildcard,
   extra,
+  capture,
   and,
-  matchers: { WHITE_SPACE: SPACE, START, LAZY },
+  or,
+  matchers: { ANY, WHITE_SPACE: SPACE, START, LAZY },
   flags
 } = require('rexrex')
 
 const space = extra(SPACE)
 const isCommandReTemplate = and(
   START, // ^
   wildcard(SPACE),
-  ['Runs' + LAZY, 'tasks' + LAZY].join(space)
+  [
+    'Runs' + LAZY,
+    'tasks' + LAZY,
+    capture(extra(ANY)),
+    capture(or('after', 'before')),
+    'this'
+  ].join(space),
+  capture(
+    or(
+      and(space, 'and', space, 'watch', space, capture(extra(ANY))),
+      and(space, 'in', space, 'parallel')
+    )
+  ) + LAZY,
+  capture(and(space, 'and', space, 'watch', space, capture(extra(ANY)))) + LAZY,
+  wildcard(SPACE)
 )
 
+const isParallelGroup = regexGroupString =>
+  /in\s+parallel/i.test(regexGroupString)
+const isWatchGroup = regexGroupString =>
+  /(.*?)and watch(.*?)/.test(regexGroupString)
 const MAID_TASKS_SECTION = whole(['<!--', 'maid-tasks', '-->'].join(space))
 
 const isCommandRe = new RegExp(isCommandReTemplate, flags.INSENSITIVE)
 const commandsRe = new RegExp(/`([^`]+)`/g)
-const isCommand = v =>
-  Boolean(v.match(isCommandRe)) && Boolean(v.match(commandsRe))
+
+// add default before this' if no 'when' is given to match default regexGroups
+const fixCommand = v =>
+  isCommandRe.test(v) && commandsRe.test(v)
+    ? v
+    : v
+        .split('and watch')[0]
+        .trim()
+        .replace(
+          /(.+)(`(.+?)`)(.+)?$/,
+          (s, $1, $2) =>
+            `${$1}${$2} before this${
+              s.split(/(.+)(`(.+?)`)(.+)$/).filter(g => g.trim() !== '')[
+                s.split(/(.+)(`(.+?)`)(.+)$/).filter(g => g.trim() !== '')
+                  .length - 1
+              ] === s
+                ? ''
+                : s.split(/(.+)(`(.+?)`)(.+)$/).filter(g => g.trim() !== '')[
+                    s.split(/(.+)(`(.+?)`)(.+)$/).filter(g => g.trim() !== '')
+                      .length - 1
```

Comment:
🙈

Reply:
:joy: I'm sorry, you have a better approach? :) @zephraph

Reply:
Ha, s'all good. Hmmm... maybe. Let me poke around a bit.

Reply:
:+1:

Reply:
fixed!